diff --git a/src/cdemud-error.c b/src/cdemud-error.c
index eb5e34c..1d0c134 100644
--- a/src/cdemud-error.c
+++ b/src/cdemud-error.c
@@ -110,13 +110,13 @@ void cdemud_error (gint errcode, GError **error) {
     gint i;
     for (i = 0; i < G_N_ELEMENTS(errors); i++) {
         if (errors[i].errcode == errcode) {
-            g_set_error(error, CDEMUD_ERROR, errors[i].errcode, errors[i].errstring);
-            return;
+            g_set_error(error, CDEMUD_ERROR, errors[i].errcode, "%s", errors[i].errstring);
+	    return;
         }
     }
 
     /* Generic error */
-	g_set_error(error, CDEMUD_ERROR, errors[i-1].errcode, errors[i-1].errstring);
+	g_set_error(error, CDEMUD_ERROR, errors[i-1].errcode, "%s", errors[i-1].errstring);
 
     return;
 }
